flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1088,6 +1091,7 @@ private TaskExecutorJobServices registerNewJobAndCreateServices(
 
         return TaskExecutorJobServices.create(
                 libraryCacheManager.registerClassLoaderLease(jobId),
+                StateChangelogStorageLoader.load(taskManagerConfiguration.getConfiguration()),
```

Comment:
We are using TM configuration here, not the job one, right?
While changelog storage can be configured per job.

So the right place to do this I think is `TaskExecutor.submitTask` - where we have access to job configuration (from task deployment descriptor).
That means we'll have to cache it like:
```
changelogStorageMap.computeIfAbsent(
        jobId,
        unused -> changelogStorageLoader.load(jobInformation.getJobConfiguration())
);
```
That also means we don't need to add it to `JobTable...` classes.

WDYT?

cc: @tillrohrmann